src/uucore/src/lib/features/backup_control.rs

```diff
@@ -416,25 +416,34 @@ pub fn get_backup_path(
 }
 
 fn simple_backup_path(path: &Path, suffix: &str) -> PathBuf {
-    let mut p = path.to_string_lossy().into_owned();
-    p.push_str(suffix);
-    PathBuf::from(p)
+    match path.file_name() {
+        Some(file_name) => {
+            let mut file_name = file_name.to_os_string();
+            file_name.push(suffix);
+            path.with_file_name(file_name)
+        }
+        None => path.with_file_name(suffix),
+    }
 }
 
 fn numbered_backup_path(path: &Path) -> PathBuf {
+    let file_name = path.file_name().unwrap_or_default();
     for i in 1_u64.. {
-        let path_str = &format!("{}.~{}~", path.to_string_lossy(), i);
-        let path = Path::new(path_str);
+        let mut numbered_file_name = file_name.to_os_string();
+        numbered_file_name.push(format!(".~{}~", i));
+        let path = path.with_file_name(numbered_file_name);
         if !path.exists() {
-            return path.to_path_buf();
+            return path;
         }
     }
     panic!("cannot create backup")
 }
 
 fn existing_backup_path(path: &Path, suffix: &str) -> PathBuf {
-    let test_path_str = &format!("{}.~1~", path.to_string_lossy());
-    let test_path = Path::new(test_path_str);
+    let file_name = path.file_name().unwrap_or_default();
+    let mut numbered_file_name = file_name.to_os_string();
+    numbered_file_name.push(".~1~");
```

Comment:
can it be incremental ? like .~2~, .~3~, etc

Reply:
I'm trying to keep the logic the same as before. I think this just checks if the numbered backup exists. Only checking for '.~1~' is enough. If it exists, then we're still using the numbered backup, else we're using simple backup.

Reply:
ping ?

Reply:
> ping ?

sorry, English isn't my native language. What does "ping" mean?

Reply:
I've noticed that there are some compilation errors being reported currently. If you intend to merge this PR, I can go and handle them.

Reply:
@hamflx: This is unrelated to your PR.